### PR TITLE
Expand merged cells to fill the range

### PR DESCRIPTION
**Pull request by tmjbradley**
When reading a worksheet with merged cells the default behaviour is to have the value in the top left cell and the remaining cells be empty. Being able to expand the merged cells to duplicate the value into all cells in the range can be helpful.

### CODE REVIEW

File: lib/roo/excelx/sheet_doc.rb [hunk 1]
```diff
@@ -121,10 +122,30 @@ def extract_hyperlinks(relationships)
     end
 
     def extract_cells(relationships)
-      Hash[doc.xpath("/worksheet/sheetData/row/c").map do |cell_xml|
+      if @options[:expand_merged_cells]
```
Comment (simonoff) [inline lib/roo/excelx/sheet_doc.rb]: Can you extract this block into separate method?

File: lib/roo/excelx/sheet_doc.rb [hunk 2]
```diff
@@ -121,10 +122,30 @@ def extract_hyperlinks(relationships)
...
+        # Extract merged cells
+        merges = {}
+        doc.xpath("/worksheet/mergeCells/mergeCell").each do |mergecell_xml|
+          tl, br = mergecell_xml['ref'].split(/:/).map {|ref| ::Roo::Utils.ref_to_key(ref)}
+          for row in tl[0]..br[0] do
+            for col in tl[1]..br[1] do
+              next if row == tl[0] and col == tl[1]
```
Comment (simonoff) [inline lib/roo/excelx/sheet_doc.rb]: Can you use `&&` instead of `and`?

**Comment by simonoff** [comment]
Would be great to have some test case for this new feature.

**Comment by tmjbradley** [comment]
Added tests with the option set and unset, and refactored out into a separate method.

**Comment by pittyacg** [comment]
Hey. Will this be usable with roo?

**Comment by tmjbradley** [comment]
It's usable from my fork, it's just pending integration. Simonoff would need to comment on that.

**Comment by pittyacg** [comment]
Thanks. Working great. :)

**Comment by dan-corneanu** [comment]
I think we should update the README file too and document the new :expand_merged_ranges.
I needed this feature recently and found no mention anywhere in the documentation.

Along the same lines,
if :expand_merged_ranges is not enabled, how can one read an xlsx file and retrieve information about what are the defined merged ranges in a sheet?

**Pull request merged**